core/src/main/java/com/homeaway/streamingplatform/db/dao/impl/KafkaManagerImpl.java

```diff
@@ -54,7 +58,7 @@
      * @param replicationFactor replication for each topic that will be created
      * @param properties        properties that will be set on each topic in the list
      */
-    public void upsertTopics(Collection<String> topics, int partitions, int replicationFactor, Properties properties) {
+    public void upsertTopics(Collection<String> topics, int partitions, int replicationFactor, Properties properties, boolean isNewStream) {
```

Comment:
Does it make sense to separate updateTopic from createTopic here so that we don't have to overload two method logic bodies into the same signature ?